app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsPresenter.java

```diff
@@ -0,0 +1,195 @@
+package com.asfoundation.wallet.ui.iab;
+
+import android.os.Bundle;
+import com.appcoins.wallet.bdsbilling.Billing;
+import com.appcoins.wallet.bdsbilling.repository.BillingSupportedType;
+import com.appcoins.wallet.bdsbilling.repository.entity.Purchase;
+import com.appcoins.wallet.bdsbilling.repository.entity.Transaction;
+import com.appcoins.wallet.billing.BillingMessagesMapper;
+import com.appcoins.wallet.billing.repository.entity.TransactionData;
+import com.asfoundation.wallet.billing.analytics.BillingAnalytics;
+import com.asfoundation.wallet.entity.TransactionBuilder;
+import com.asfoundation.wallet.repository.BdsPendingTransactionService;
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.Scheduler;
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class PaymentMethodsPresenter {
+  private final PaymentMethodsView view;
+  private final Scheduler viewScheduler;
+  private final Scheduler networkThread;
+  private final CompositeDisposable disposables;
+  private final InAppPurchaseInteractor inAppPurchaseInteractor;
+
+  private final String appPackage;
+  private final BillingMessagesMapper billingMessagesMapper;
+  private final BdsPendingTransactionService bdsPendingTransactionService;
+  private final Billing billing;
+  private final BillingAnalytics analytics;
+  private final boolean isBds;
+  private final Single<TransactionBuilder> transactionBuilder;
+  private final String developerPayload;
+  private final String uri;
+
+  public PaymentMethodsPresenter(PaymentMethodsView view, String appPackage,
+      Scheduler viewScheduler, Scheduler networkThread, CompositeDisposable disposables,
+      InAppPurchaseInteractor inAppPurchaseInteractor, BillingMessagesMapper billingMessagesMapper,
+      BdsPendingTransactionService bdsPendingTransactionService, Billing billing,
+      BillingAnalytics analytics, boolean isBds, Single<TransactionBuilder> transactionBuilder,
+      String developerPayload, String uri) {
+    this.view = view;
+    this.appPackage = appPackage;
+    this.viewScheduler = viewScheduler;
+    this.networkThread = networkThread;
+    this.disposables = disposables;
+    this.inAppPurchaseInteractor = inAppPurchaseInteractor;
+    this.billingMessagesMapper = billingMessagesMapper;
+    this.bdsPendingTransactionService = bdsPendingTransactionService;
+    this.billing = billing;
+    this.analytics = analytics;
+    this.isBds = isBds;
+    this.transactionBuilder = transactionBuilder;
+    this.developerPayload = developerPayload;
+    this.uri = uri;
+  }
+
+  public void present(double transactionValue, String currency, Bundle savedInstanceState) {
+    if (savedInstanceState == null) {
```

Comment:
be careful with the lifecycle. In this case disposables are subscribed on fist time the view is created and disposed when view is destroyed. Here If we recreate the view, click listeners are not registered